File: Courses/Intro-to-Comp-Sci/labs/map&reduce/lab1.py
############################################################
# Name: Adib Osmany
# Pledge: I pledge my honor that I have abided by the Stevens Honor System.
# CS115 Lab 1
#  
############################################################
from math import factorial
from functools import reduce

def inverse(x):
    '''returns the reciprocal of x'''
    return (1/x)

'''this function is meant to be used with the reduce with list3
to help find the factorial of the given number'''
def multi(z,t):
    '''returns the product of z and t'''
    return z*t


def factorial(y):
    '''returns the factorial of y'''
    list3=(range(1,y+1))
    return reduce(multi,(list3))
    
def e(n):
    list1=list(range(1,n+1))
    list4=list(map(factorial,(list1)))
    list2=list(map(inverse,(list4)))
    return sum(list2)+1
    
    

File: Computer-Architecture/labs/Binary/lab1.c
 /*******************************************************************************
 * Author  : Adib Osmany
 * Pledge  : I pledge my honor that I have abided by the Stevens Honor System.
 ******************************************************************************/
 #include <stdio.h>
 #include <stdlib.h>

 void display(int8_t bit) {
    putchar(bit + 48);
 }

 void display_32(int32_t num) {
    int i = 31;
    while(i>=0){
        if (((num >> i) & 1) == 0){
            display(0);
        }
        else {
            display(1);
        }
        i--;
    }
    
 }

 int main(int argc, char const *argv[]) {
      display_32(382);
      return 0;
  }